### PR TITLE
MDEV-18136 Server crashes in Item_func_dyncol_create::prepare_arguments

### PR DESCRIPTION
**Pull request by midenok**
This code is submitted under the BSD-new license.

### CODE REVIEW

File: sql/partition_info.h [hunk 1]
```diff
@@ -395,16 +395,25 @@ class partition_info : public Sql_alloc
...
     vers_info->interval.start= start;
-    return get_interval_value(item, int_type, &vers_info->interval.step) ||
+    if (item->fix_fields_if_needed_for_scalar(thd, &item))
+      return true;
+    bool error= get_interval_value(item, int_type, &vers_info->interval.step) ||
            vers_info->interval.step.neg || vers_info->interval.step.second_part ||
           !(vers_info->interval.step.year || vers_info->interval.step.month ||
             vers_info->interval.step.day || vers_info->interval.step.hour ||
             vers_info->interval.step.minute || vers_info->interval.step.second);
+    if (error) {
```
Comment (an3l) [inline sql/partition_info.h]: Maybe new line for `{` ?
Can you explain me why is this part moved here and not in yacc ?
Thanks

Comment (midenok) [reply sql/partition_info.h]: I guess it eliminates some code duplication.

**Comment by abarkov** [comment]
Merged to MariaDB-10.3.6
